lib/matplotlib/__init__.py

```diff
@@ -995,6 +995,35 @@ def rc_params_from_file(fname, fail_on_error=False, use_default_template=True):
     locale.setlocale(locale.LC_ALL, '')
 
 
+def _run_submodule_inits():
+    """
+    Initialize matplotlib submodules that require deferred initialization.
+
+    This function is called once after rcParams have been loaded, providing
+    a centralized orchestration point for all submodule initialization logic.
+    This decouples module imports from initialization, making the codebase
+    more maintainable and less fragile to import order issues.
+
+    Submodules that need mandatory initialization should implement a
+    ``_mpl_init()`` function, which will be called by this orchestrator.
+
+    See Also
+    --------
+    Issue #29813 : Cleanup internal import dependencies and initialization logic
+    """
+    # Import and initialize submodules in dependency order
+    # Each submodule's _mpl_init() is responsible for its own setup
+
+    # Initialize colormap registry (depends on rcParams['image.lut'])
+    from matplotlib import cm
+    if hasattr(cm, '_mpl_init'):
+        cm._mpl_init()
+
+
+# Run submodule initialization after rcParams are loaded
+_run_submodule_inits()
```

Comment:
might as well put this almost at the bottom?  The thing in particular that gets us in trouble is that if we access an attribute on `matplotlib` in a sub-module before we have created it.  `rcParams` is the most common problem (and the problem here) but we should cover the (more) general case and put this call as late as possible.